clients/drcachesim/analyzer.cpp

```diff
@@ -228,17 +236,26 @@ analyzer_tmpl_t<RecordType, ReaderType>::init_scheduler_common(
     }
     std::vector<typename sched_type_t::input_workload_t> sched_inputs(1);
     sched_inputs[0] = std::move(workload);
+
     typename sched_type_t::scheduler_options_t sched_ops;
-    int output_count;
-    if (parallel_) {
+    if (shard_type_ == SHARD_BY_CORE) {
+        // Subclass must pass us options and set worker_count_ to # cores.
```

Comment:
For a dynamic schedule we need each core to execute simultaneously with the others.  If we want to map multiple cores onto one worker we have to do a lockstep multiplex or something: we cannot just run one core to "completion" and then switch to another, like we can do with threads.  I think it is nice and simple to have one core per worker.  Please elaborate on why you think this is restrictive?

Reply:
I realize I had a "static core schedule" model in mind, where all core-shards can be processed completely independently. I understand the "dynamic" part now.

I wonder if there are any assumptions in the analyzer framework that would break if the different workers rely on each other making steady progress (before, we had only one join point, the one at the end). I assume we'll do testing on large traces before submitting this?

Reply:
I added to the docs a mention of the timing mattering and that replay mode can be used to work around problems with asymmetric overhead tools.  Will leave unresolved until some testing on large traces is done -- which is my own main initial use case, to convert schedule_launcher into an analysis tool and run it on large traces.

Reply:
Ran on a larger trace and it seems fine:
Thread-sharded has >400 threads:
```
Total counts:                      
  1269911279 total (fetched) instructions
Thread 972956 counts:              
    15371856 (fetched) instructions
Thread 977148 counts:              
    13345146 (fetched) instructions
Thread 973164 counts:              
    12826655 (fetched) instructions
...
          17 (fetched) instructions
Thread 972835 counts:
          17 (fetched) instructions
Thread 972805 counts:
          17 (fetched) instructions
```

Core-sharded with 8 cores is fairly well balanced:
```
Total counts:
  1269911279 total (fetched) instructions
Core 7 counts:
   189906370 (fetched) instructions
Core 0 counts:
   183510188 (fetched) instructions
Core 4 counts:
   171494771 (fetched) instructions
Core 3 counts:
   162365542 (fetched) instructions
Core 2 counts:
   159228453 (fetched) instructions
Core 5 counts:
   141998485 (fetched) instructions
Core 6 counts:
   134594713 (fetched) instructions
Core 1 counts:
   126812757 (fetched) instructions
```